packages/transformers/typescript-tsc/src/TSCTransformer.js

```diff
@@ -0,0 +1,82 @@
+// @flow strict-local
+
+import {Transformer} from '@parcel/plugin';
+import localRequire from '@parcel/local-require';
+
+type TypescriptCompilerOptions = {
+  module?: mixed,
+  jsx?: mixed,
+  noEmit?: boolean,
+  sourceMap?: boolean
+};
+
+type TypescriptTranspilerOptions = {
+  compilerOptions: TypescriptCompilerOptions,
+  fileName: string
+};
+
+export default new Transformer({
+  async getConfig({asset}) {
+    return asset.getConfig(['tsconfig.json']);
+  },
+
+  async transform({asset, config}) {
+    asset.type = 'js';
+
+    let typescript =
+      config == null
+        ? // $FlowFixMe
+          require('typescript')
+        : await localRequire('typescript', asset.filePath);
+
+    // require typescript, installed locally in the app
+    let transpilerOptions: TypescriptTranspilerOptions = {
+      compilerOptions: {
+        // Don't compile ES `import`s -- scope hoisting prefers them and they will
+        // otherwise compiled to CJS via babel in the js transformer
+        module: typescript.ModuleKind.ESNext,
+
+        // React is the default. Users can override this by supplying their own tsconfig,
+        // which many TypeScript users will already have for typechecking, etc.
+        jsx: 'React'
+      },
+      fileName: asset.filePath // Should be relativePath?
+    };
+
+    // Overwrite default if config is found
+    if (config) {
+      transpilerOptions.compilerOptions = {
+        ...transpilerOptions.compilerOptions,
+        ...config.compilerOptions
```

Comment:
`compilerOptions.module` should not be overwritten by the user config if scope hoisting is enabled